toolchain/check/call.cpp

```diff
@@ -141,10 +143,20 @@ auto PerformCall(Context& context, SemIR::LocId loc_id, SemIR::InstId callee_id,
   // for the call.
   auto callee_specific_id = ResolveCalleeInCall(
       context, loc_id, callable, "function", callable.generic_id,
-      callee_function.specific_id, callee_function.self_id, arg_ids);
+      callee_function.enclosing_specific_id, callee_function.self_id, arg_ids);
   if (!callee_specific_id) {
     return SemIR::InstId::BuiltinError;
   }
+  if (callee_specific_id->is_valid()) {
```

Comment:
What does it mean for `callee_specific_id` to be invalid, as opposed to null?

Reply:
An invalid `SpecificId` generally means "not generic". The nullopt return value here is used to indicate that we're recovering from an error.

Reply:
OK, this is a bit of a drive-by fix, but could you document that on `ResolveCalleeInCall`?

Reply:
Done.